### PR TITLE
Support version 3 of `MathJax` and add `config.typesetMath` 

### PR DESCRIPTION
**Pull request by archmoj**
TODOs: 
- [x] dashboard test using `npm run mathjax3`
- [x] dashboard example using `npm run mathjax3chtml` when need to output `chtml` on the page as well as `svg` output for plotly graphs | thanks to @akhmerov
- [x] jasmine tests
- [x] add image test using mathjax-v3
- [x] documentation

@plotly/plotly_js

### CODE REVIEW

**Comment by akhmerov** [comment]
Note that this won't work if the page mathjax3 uses a chtml output plugin and AFAIU most people prefer chtml. A workaround requires creating a separate virtual document for MathJax to use in typesetting, see https://github.com/plotly/plotly.js/issues/4563#issuecomment-1004029245 for a prototype implementation.

File: src/lib/svg_text_utils.js [hunk 1]
```diff
@@ -196,47 +203,83 @@ function texToSVG(_texString, _config, _callback) {
...
-        return MathJax.Hub.Typeset(tmpDiv.node());
-    },
-    function() {
+        var tmpNode = tmpDiv.node();
+
+        if(MathJaxVersion < 3) {
+            return MathJax.Hub.Typeset(tmpNode);
+        } {
+            return MathJax.typeset([tmpNode]);
+        }
```
Comment (alexcjohnson) [inline src/lib/svg_text_utils.js:230]: Interesting, I didn't know you could make a block like this second one, ie with no control statement preceding it. I suppose there's a nice symmetry to this, as the two typeset statements are equally indented, but it's confusing to read so I'd rather we remove it.
```suggestion
        if(MathJaxVersion < 3) {
            return MathJax.Hub.Typeset(tmpNode);
        }
        return MathJax.typeset([tmpNode]);
```

**Comment by akhmerov** [comment]
Another regression is that compared to the v2 API this doesn't preserve the MathJax settings.

**Comment by archmoj** [comment]
@akhmerov thanks very much for the notes.
Please see https://github.com/plotly/plotly.js/pull/6073/commits/4ba005f80cff5acee302642534942f0f38698a40.
With your workaround now one could use `npm run mathjax3chtml` to test `chtml` with `svg` on the same page when `_mockedV2API` flag is set to true on the `MathJax` object.

**Comment by akhmerov** [comment]
Thanks! Note that you don't actually need to mock v2 API to combine the outputs. I mocked the API to make it work with the current plotly implementation and combined with a second virtual document from another recipe. Instead you'll want to use just the recipe from https://github.com/mathjax/MathJax/issues/2705#issuecomment-859742446.

**Comment by archmoj** [comment]
> Thanks! Note that you don't actually need to mock v2 API to combine the outputs. I mocked the API to make it work with the current plotly implementation and combined with a second virtual document from another recipe. Instead you'll want to use just the recipe from [mathjax/MathJax#2705 (comment)](https://github.com/mathjax/MathJax/issues/2705#issuecomment-859742446).

Good news. Let me fix that part then :)

**Comment by archmoj** [comment]
@akhmerov the PR is ready for your review :)

**Comment by akhmerov** [comment]
At a cursory read looks very reasonable. I am a bit concerned if calling `defaultReady` multiple times has side effects, but I don't know MathJax enough to answer this.

**Comment by akhmerov** [comment]
@archmoj please see also this comment by @dpvc https://github.com/mathjax/MathJax/issues/2705#issuecomment-1007748934: it's more efficient to avoid DOM manipulation and use `tex2svgPromise` directly (although this still requires a different virtual document).

File: README.md [hunk 2]
```diff
@@ -87,11 +87,19 @@ While non-minified source files may contain characters outside UTF-8, it is reco
...
+To support MathJax, you could load either version two or version three of MathJax files, for example:
 ```html
 <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-AMS-MML_SVG.js"></script>
 ```
 
+```html
+<script src="https://cdn.jsdelivr.net/npm/mathjax@3.2.0/es5/tex-svg.js"></script>
+```
+
+> When using MathJax version 3, it is also possibly to use `chtml` output on the other parts of the page in addition to `svg` output for the plotly graph.
```
Comment (alexcjohnson) [inline README.md:99]:
```suggestion
> When using MathJax version 3, it is also possible to use `chtml` output on the other parts of the page in addition to `svg` output for the plotly graph.
```

File: devtools/test_dashboard/index-mathjax3chtml.html [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  <p>MathJax $V^3$ with $chtml$ output on the page and <b>svg</b> output on the plotly graphs</p>
+  <div id="plots">
+    <div id="graph"></div>
+  </div>
+  <div id="snapshot"></div>
+
+  <script>
+    window.MathJax = window.MathJax || {
+      // see https://github.com/mathjax/MathJax/issues/2705#issuecomment-859742446
+      loader: {load: ['output/svg']},
```
Comment (alexcjohnson) [inline devtools/test_dashboard/index-mathjax3chtml.html:28]: interesting... I guess if this becomes a common use case we could try to bake similar code (but designed to run later, not on startup) into plotly.js directly. For now though this is fine, as long as we're comfortable using only svg rendering in dash markdown.

File: package.json [hunk 4]
```diff
@@ -52,6 +52,8 @@
     "test-requirejs": "node tasks/test_requirejs.js",
     "test-plain-obj": "node tasks/test_plain_obj.js",
     "test": "npm run test-jasmine -- --nowatch && npm run test-bundle && npm run test-image && npm run test-export && npm run test-syntax && npm run lint",
+    "mathjax3": "node devtools/test_dashboard/server.js --mathjax3",
+    "mathjax3chtml": "node devtools/test_dashboard/server.js --mathjax3chtml",
```
Comment (alexcjohnson) [inline package.json:56]: 🌴  slightly cleaner (also for `strict`): `npm run start -- --mathjax3`

File: src/lib/svg_text_utils.js [hunk 5]
```diff
@@ -173,70 +173,149 @@ function cleanEscapesForTex(s) {
...
+                tex2jax: {
+                    inlineMath: inlineMath
+                },
+                displayAlign: 'left',
+            });
+        } else {
+            originalConfig = Lib.extendDeepAll({}, MathJax.config);
+
+            if(!MathJax.config.tex.inlineMath) {
+                MathJax.config.tex.inlineMath = inlineMath;
```
Comment (alexcjohnson) [inline src/lib/svg_text_utils.js:213]: So if `inlineMath` is already set some other way we would honor that? Seems risky, means the graph would behave How about we always use our own, but if at some point we want to allow other delimiters we make a layout attr to provide them within the graph?

File: src/lib/svg_text_utils.js [hunk 6]
```diff
@@ -173,70 +173,149 @@ function cleanEscapesForTex(s) {
...
     var originalRenderer,
         originalConfig,
         originalProcessSectionDelay,
         tmpDiv;
 
-    MathJax.Hub.Queue(
-    function() {
-        originalConfig = Lib.extendDeepAll({}, MathJax.Hub.config);
+    var setConfig = function() {
+        if(MathJaxVersion < 3) {
```
Comment (alexcjohnson) [inline src/lib/svg_text_utils.js:193]: Some of these functions don't share anything internal between 2 and 3, and they get invoked separately for the two cases - might be cleaner and more readable to make separate functions `setConfig2` and `setConfig3`.

**Comment by alexcjohnson** [comment]
Looks good! Only the question about `inlineMath` delimiters is blocking.

@akhmerov thanks for the reviews! Re: `tex2svgPromise`, I think we should merge with the current structure, but if performance becomes a concern it's good to know this is an option.

File: src/lib/svg_text_utils.js (hunk 5 above)
Comment (archmoj) [reply src/lib/svg_text_utils.js:215]: Addressed in https://github.com/plotly/plotly.js/pull/6073/commits/1de176ca87c125a1704b76f4f56be18409f02a2b.

File: src/lib/svg_text_utils.js (hunk 6 above)
Comment (archmoj) [reply src/lib/svg_text_utils.js]: Good call. Addressed in https://github.com/plotly/plotly.js/pull/6073/commits/7677258b8ef5a1579167cbfc84bcbd7ade94311e.

File: package.json (hunk 4 above)
Comment (archmoj) [reply package.json:56]: Thanks for the suggestion. But let's skip this for now.

**Comment by archmoj** [comment]
@alexcjohnson this PR is pretty much ready for the final review.
Thank you!

File: src/plot_api/plot_config.js [hunk 7]
```diff
@@ -22,6 +22,12 @@ var configAttributes = {
         ].join(' ')
     },
 
+    typesetMath: {
+        valType: 'boolean',
+        dflt: true,
+        description: 'Determines whether the math should be typeset or not.'
```
Comment (alexcjohnson) [inline src/plot_api/plot_config.js:28]:
```suggestion
        description: [
            'Determines whether math should be typeset or not,',
            'when MathJax (either v2 or v3) is present on the page.'
        ].join(' ')
```

**Review by alexcjohnson — APPROVED** [review]
💃 Very nice work!

File: src/lib/svg_text_utils.js (hunk 5 above)
Comment (alexcjohnson) [reply src/lib/svg_text_utils.js]: With https://github.com/plotly/plotly.js/pull/6073/commits/d5e10a7de0ab3ad4f8c83f7745a8101abcce221d looks great 😄

**Pull request merged**